src/main/java/io/supertokens/passwordless/Passwordless.java

```diff
@@ -409,6 +410,20 @@ public static ConsumeCodeResponse consumeCode(TenantIdentifier tenantIdentifier,
             }
             throw e;
         }
+    }
+
+    public static ConsumeCodeResponse consumeCode(TenantIdentifier tenantIdentifier, Storage storage, Main main,
```

Comment:
the diff makes it hard to see, but, is the only change in this function only that you are calling checkCodeAndReturnDevice at the start instead of doing the code consume transaction in this function itself? What happens after that remains the same right?

Reply:
Technically there is no code change in the consume code function. First part where the code check was happening is refactored into a separate function. only change is that we pass a flag to decide whether we want to delete the device on success or not